store/store.go

```diff
@@ -332,8 +334,70 @@ func authenticate(r *http.Request, user *auth.UserState) {
 	r.Header.Set("Authorization", buf.String())
 }
 
+// refreshMacaroon will request a refreshed discharge macaroon for the user
+func refreshMacaroon(user *auth.UserState) error {
+	for i, d := range user.StoreDischarges {
+		discharge, err := MacaroonDeserialize(d)
+		if err != nil {
+			return err
+		}
+		if discharge.Location() == UbuntuoneLocation {
+			refreshedDischarge, err := RefreshDischargeMacaroon(d)
+			if err != nil {
+				return err
+			}
+			user.StoreDischarges[i] = refreshedDischarge
+		}
+	}
+	return nil
+}
+
+// doStoreRequest does an authenticated request to the store handling a potential macaroon refresh required if needed
+func (s *Store) doStoreRequest(client *http.Client, method, urlStr, accept string, data []byte, user *auth.UserState) (*http.Response, error) {
```

Comment:
s/doStoreRequest/doRequest/ ?

This is all about the *Store, and we have newRequest already. Seems like a good fit.